### PR TITLE
Documentation for smarthost.py

### PR DESCRIPTION
**Pull request by stephdl**
See #261

### CODE REVIEW

File: docs/core/smarthost.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+smtp = sp.load_smtp()
+json.dump(smtp, fp=sys.stdout)
+```
+
+output example:
+```json
+{"port": 587, "host": "", "username": "", "password": "", "enabled": false, "tls": true, "tls_verify": true}
+```
+
+- tls: enable startTls (true/false)
```
Comment (DavidePrincipi) [inline docs/core/smarthost.md:33]:
```suggestion
- tls: enable STARTTLS (true/false)
```

File: docs/core/smarthost.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- tls_verify: verify if the certificate is valid and if the hostname is associated to the certificate (true/false)
+
+
+The module can handle account provider changes by defining an event
+handler. Create an executable script with path
+`${AGENT_INSTALL_DIR}/events/smarthost-provider-changed/10handler` and run
```
Comment (DavidePrincipi) [inline docs/core/smarthost.md:39]: What about a shorter name?

```suggestion
`${AGENT_INSTALL_DIR}/events/smarthost-changed/10handler` and run
```

File: docs/core/smarthost.md (hunk 1 above)
Comment (stephdl) [reply docs/core/smarthost.md:33]: done

File: docs/core/smarthost.md (hunk 2 above)
Comment (stephdl) [reply docs/core/smarthost.md:39]: done

File: docs/core/smarthost.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+output example:
+```json
+{"port": 587, "host": "", "username": "", "password": "", "enabled": false, "tls": true, "tls_verify": true}
+```
+
+- tls: enable STARTTLS (true/false)
+- tls_verify: verify if the certificate is valid and if the hostname is associated to the certificate (true/false)
+
+
+The module can handle account provider changes by defining an event
```
Comment (DavidePrincipi) [inline docs/core/smarthost.md:37]:
```suggestion
The module can handle smarthost setup changes by defining an event
```

**Review by DavidePrincipi — APPROVED** [review]

**Comment by DavidePrincipi** [comment]
Recall to update the docs with latest changes from #261 

I think it could be also useful to write down a code example that answers this question: is the smarthost completely disabled (or unconfigured)?

**Review by DavidePrincipi — APPROVED** [review]
Ok, I guess to check if the smarthost is enabled I have to do

```python
smtp = agent.get_smarthost_settings(agent.redis_connect())
if smtp['enabled']:
    pass
```

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**